[Feature 42] Build profile screen with the "edit info" features

alexxispn opened the pull request:
## Description

Build profile screen with feature of edit email and edit password

## Changes

- Rename Icons, so they all have Icon at the end of their names.
- Update AppBackButton to accept property toGo to navigate to another View.
- Update mock and types of User so they can be used on Profile Screen.
- Create new Hook for show/togglePassword and update LoginForm and useLoginForm for this change.
- Add navigation to PasswordRecoveryForm so now when u complete the password recovery form you navigate to Login View.
- Add custom hooks and Forms for EditEmail, EditPrevPassword and EditNewPassword.
- Add pages for profile screen, edit email, confirm old password, edit new password.

## Checks

- [X] Project Builds
- [X] Project passes tests and checks
- [X] Updated documentation accordingly
- [X] Back arrow in top left corner leads to burger menu (screen 45)
- [X] Screen follows provided designs

## Additional information

Reference: [42] Build profile screen with the "edit info" features
URL: https://github.com/TheTributeCommunity/fesbal-frontend/issues/26


This branch was created from the branch 'feature-46-47/notification-screen-plus-notifications-detailed-view' in order not to have to redo tasks already done.
https://github.com/TheTributeCommunity/fesbal-frontend/tree/feature-46-47/notification-screen-plus-notifications-detailed-view

albertopolofer commented on src/types/UserProps.ts:
```diff
@@ -0,0 +1,15 @@
+}
+
+type FamilyMember = {
+    FullName: string;
```
minor case typo

albertopolofer commented:
Awesome job so far @alexxispn ! 🚀 

And I'm very impressed to see some minor issues I've seen in the previous pr have been fixed.

Just one thing is that in the DoD of the issue it says: 
Back arrow in top left corner leads to burger menu (screen 45)

As far as I've seen there is no such page yet, so it makes sense it takes me to login screen again, so it's fine.

alexxispn commented:
Hi @albertopolofer and @braisgonzalez 

I have made the following changes based on your feedback on my previous pull request:
- Installed FontAwesome and updated all views and components with FontAwesome icons
- Installed and configured i18next, and updated pages and components. Created an i18n folder with English and Spanish translations. Updated all static text on views to use the i18n Spanish translation.
- Removed styles for SweetAlert to remove all unnecessary "!important" declarations
- Created a `src/styles/variables.scss` file and adapted all static classes on Tailwind to use variables from `variables.scss`

I appreciate your guidance and the time you took to review my previous pull request. Your feedback has helped me a lot to improve the codebase and make it more maintainable.

Please let me know if there's anything else I should take into account before it's ready for a final review.
